hybrasyl/Objects/User.cs

```diff
@@ -1260,35 +1260,174 @@ public void SendVisibleGold(Gold gold)
         internal void UseSkill(byte slot)
         {
             var castable = SkillBook[slot];
-
-            Attack(castable);
-
-            Client.Enqueue(new ServerPacketStructures.Cooldown()
+            if (canCast(castable))
             {
-                Length = (uint)castable.Cooldown,
-                Pane = 1,
-                Slot = slot
-            }.Packet());;
+                bool spellCast = false;
+                foreach (var tar in PossibleTargets(castable))
+                {
+                    bool canAttack = !(tar is Merchant || (tar is User && (!(tar as User).Status.HasFlag(PlayerCondition.Pvp))));
+                    if (canAttack)
+                    {
+                        Attack(castable, tar);
+                        spellCast = true;
+                    }
+                }
+                if (spellCast) CastingCost(castable);
+
+                Client.Enqueue(new ServerPacketStructures.Cooldown()
+                {
+                    Length = (uint)castable.Cooldown,
+                    Pane = 1,
+                    Slot = slot
+                }.Packet()); ;
+            }
         }
 
         internal void UseSpell(byte slot, uint target = 0)
         {
             var castable = SpellBook[slot];
-            Creature targetCreature = Map.EntityTree.OfType<Monster>().SingleOrDefault(x => x.Id == target) ?? null;
-            Direction playerFacing = this.Direction;
-            if (targetCreature is Merchant) return;
-            if (targetCreature != null) Attack(castable, targetCreature);
-            else Attack(castable);
+            Creature targetCreature = Map.EntityTree.OfType<Creature>().SingleOrDefault(x => x.Id == target) ?? null;
+            if (canCast(castable))

```

Comment:
canCast should use player status.

The attack method should handle all checks.

Reply:
Will add player status to canCast.